From 10716f648ac5f0a0f6ceae62e7ce6f3072d27310 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:24:12 -0400
Subject: [pci] Use device_release_driver() in pci_stop_root_bus()

Message-id: <20140719042411.19438.56.stgit@gir.stowe>
Patchwork-id: 85976
O-Subject: [RHEL7.1 PATCH 14/81] PCI: Use device_release_driver() in pci_stop_root_bus()
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: e3b439e1d315aff59c1b4f0fc43c5cd2d82b4138

commit e3b439e1d315aff59c1b4f0fc43c5cd2d82b4138
Author: Yinghai Lu <yinghai@kernel.org>
Date:   Sat Nov 30 14:40:27 2013 -0800

    PCI: Use device_release_driver() in pci_stop_root_bus()

    To be consistent with 4bff6749905d ("PCI: Move device_del() from
    pci_stop_dev() to pci_destroy_dev()", this changes pci_stop_root_bus()
    to use device_release_driver() instead of device_del().

    This also changes pci_remove_root_bus() to use device_unregister()
    instead of put_device() so it corresponds with the device_register()
    call in pci_create_root_bus().

    [bhelgaas: changelog]
    Signed-off-by: Yinghai Lu <yinghai@kernel.org>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Acked-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/remove.c b/drivers/pci/remove.c
index cc9337a..692f4c3 100644
--- a/drivers/pci/remove.c
+++ b/drivers/pci/remove.c
@@ -128,7 +128,7 @@ void pci_stop_root_bus(struct pci_bus *bus)
   pci_stop_bus_device(child);
 
  /* stop the host bridge */
- device_del(&host_bridge->dev);
+ device_release_driver(&host_bridge->dev);
 }
 
 void pci_remove_root_bus(struct pci_bus *bus)
@@ -147,5 +147,5 @@ void pci_remove_root_bus(struct pci_bus *bus)
  host_bridge->bus = NULL;
 
  /* remove the host bridge */
- put_device(&host_bridge->dev);
+ device_unregister(&host_bridge->dev);
 }
-- 
1.7.1